### PR TITLE
refactor(core): Move license endpoints to a decorated controller class (no-changelog)

### PR DESCRIPTION
**Pull request by netroy**
## Summary
This Pr continues the effort of migrating all endpoints to decorated controller class

## Review / Merge checklist
- [x] PR title and summary are descriptive.
- [x] Tests included.

### CODE REVIEW

File: packages/cli/src/license/license.controller.ts [hunk 1]
```diff
@@ -1,129 +1,37 @@
...
+import { LicenseService } from './license.service';
+
+@Service()
+@Authorized()
+@RestController('/license')
+export class LicenseController {
+	constructor(private readonly licenseService: LicenseService) {}
+
+	@Get('/')
+	async getLicenseData() {
```
Comment (ivov) [inline packages/cli/src/license/license.controller.ts:13]: Should we start using `@NoAuthRequired()` to make this more explicit?

File: packages/cli/test/unit/License.test.ts [hunk 2]
```diff
@@ -86,13 +86,13 @@ describe('License', () => {
 	});
 
 	test('check if feature is enabled', async () => {
-		await license.isFeatureEnabled(MOCK_FEATURE_FLAG);
+		license.isFeatureEnabled(MOCK_FEATURE_FLAG);
```
Comment (ivov) [inline packages/cli/test/unit/License.test.ts:89]: Also the `async` on L88. Same below.

File: packages/cli/src/license/license.controller.ts (hunk 1 above)
Comment (netroy) [reply packages/cli/src/license/license.controller.ts:13]: but, so far auth has been required for this endpoint. that's why there is an `@Authorized` on the class.

Comment (ivov) [reply packages/cli/src/license/license.controller.ts:13]: Ah of course!

**Review by ivov — APPROVED** [review]

**Comment by janober** [comment]
Got released with [`n8n@1.22.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.22.0)

**Pull request merged**